### ISSUE TITLE
errors

### ISSUE THREAD

**Issue by classabbyamp** [miaowware/ae7qparser#7]
to think about:
- request errors
- not found errors
    - seems that only applications, LIDs, and FRNs have 0 tables when not found
    - callsigns show a conditions table no matter what
- error on foreign?

**Comment by classabbyamp** [comment]
request errors: pass to the user
not found errors: not sure
error on foreign: I don't think so, because it will give a conditions table

**Comment by 0x5c** [comment]
Yeah, rq err and foreign seem alright to me, but imo we should not have a "not found" error; that should be left to whoever is using our API, instead just returning the tables found, no tables if none found.

**Comment by classabbyamp** [comment]
I think foreign should be considered the same as not found, i.e. it will always respond to a callsign query with a conditions table if no callsign history etc is found. That makes things very easy, either:

1. pass all request errors as-is
2. custom error class to encapsulate those errors

**Comment by 0x5c** [comment]
yeah, that's what I said

**Comment by classabbyamp** [comment]
oh I read

> Yeah, rq err and foreign seem alright to me

as "doing rq err and foreign seems alright"

**Comment by 0x5c** [comment]
Encapsulating for rq/parse error makes sense in that it allows catching all errors from the module

`Ae7qError`
- `Ae7qRequestError`
- `Ae7qParsingError`
- *etc*
